lti_consumer/plugin/compat.py

```diff
@@ -129,3 +138,25 @@ def user_course_access(*args, **kwargs):
     # pylint: disable=import-error,import-outside-toplevel
     from lms.djangoapps.courseware.courses import check_course_access
     return check_course_access(*args, **kwargs)
+
+
+def request_cached(func) -> Callable[[Callable], Callable]:
+    """
+    Import the `request_cached` decorator from LMS and apply it if available.
+    """
+    try:
+        # pylint: disable=import-outside-toplevel
+        from openedx.core.lib.cache_utils import request_cached as lms_request_cached
+        return lms_request_cached(func)
+    except ImportError:
+        log.warning("Unable to import `request_cached`. This is normal if running tests.")
+        return func
+
+
+def clean_course_id(model_form: ModelForm) -> CourseKey:
+    """
+    Import and run `clean_course_id` from LMS
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from openedx.core.lib.courses import clean_course_id as lms_clean_course_id
```

Comment:
Can you please explain why we have this import here?

Reply:
Do you mean why it's imported at all or why it's imported inside a function?
It's being imported from the LMS to avoid duplication of this code. 
It's imported inside a function because when running tests etc we don't have edx-platform available, which will cause import errors if the import is performed at the top of the file. By putting it in a function, we avoid the import happening till the function is actually called. So during testing we can mock this function, which will avoid the import.

Reply:
Got it. :) thanks.